test/pjrt/test_mesh_service.py

```diff
@@ -0,0 +1,50 @@
+import itertools
+import os
+from typing import List, Optional
+from absl.testing import absltest, parameterized
+
+import torch.distributed as dist
+import torch_xla.core.xla_model as xm
+from torch_xla.experimental import pjrt
+
+
+class PjRtMeshServiceTest(parameterized.TestCase):
+
+  @staticmethod
+  def _rendezvous_default(xrt_mesh_addr: Optional[str], replicas: List[int]):
+    if xrt_mesh_addr:
+      os.environ['XRT_MESH_SERVICE_ADDRESS'] = xrt_mesh_addr
+
+    payload = b'message %d' % xm.get_ordinal()
+    return xm.get_ordinal(), xm.rendezvous("test rendezvous", payload, replicas)
+
+  @parameterized.named_parameters(
+      ('defaults', None, []), ('xrt_address', 'localhost:9477', []),
+      ('four_replicas', None, [0, 1, 2, 3]), ('two_replicas', None, [0, 1]))
+  def test_rendezvous(self, xrt_mesh_addr, replicas):
+    results = pjrt.run_multiprocess(self._rendezvous_default, xrt_mesh_addr,
+                                    replicas)
+    values = list(
+        itertools.chain.from_iterable(row.values() for row in results.values()))
+    replicas = replicas or list(range(len(values)))
+
+    for ordinal, value in values:
+      if ordinal in replicas or not replicas:
+        self.assertEqual(value, [b'message %d' % r for r in replicas])
+
+  @staticmethod
+  def _mesh_reduce():
+    return xm.mesh_reduce('test mesh reduce', xm.get_ordinal(), sum)
+
+  def test_mesh_reduce(self):
```

Comment:
Would mesh_reduce just work? I thought we need to implement something else given that we don't use mesh server in PJRT world.

Reply:
I was surprised this just works too. `mesh_reduce` uses `rendezvous` to communicate serialized objects between replicas:

https://github.com/pytorch/xla/blob/227ac13d7102f4e30540df7ef50d09a02a985290/torch_xla/core/xla_model.py#L1118